### PR TITLE
[clang][Interp] Decay arrays to the first element

### PR DESCRIPTION
**Pull request by tbaederr**
I _think_ this makes sense to do here. This way we have a pointer to the first element on the stack.

### CODE REVIEW

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

File: clang/test/AST/Interp/arrays.cpp [hunk 1]
```diff
@@ -27,6 +27,9 @@ static_assert(foo[2][3] == &m, "");
 static_assert(foo[2][4] == nullptr, "");
 
 
+const int SomeInt[] = {1};
+int getSomeInt() { return *SomeInt; }
```
Comment (AaronBallman) [inline clang/test/AST/Interp/arrays.cpp:31]:
```suggestion
constexpr int SomeInt[] = {1};
constexpr int getSomeInt() { return *SomeInt; }

static_assert(getSomeInt() == 1);
```
so we're testing the decayed pointer.

**Review by AaronBallman — APPROVED** [review]
LGTM with a testing nit

**Pull request merged**